### PR TITLE
PCHR-2817: Fix bug in fetching webform submissions.

### PR DESCRIPTION
**Pull request by mickadoo**
## Overview

Webform submissions are checked on login. If a user has not completed the onboarding webform they will be redirected to it. There is a bug in submission checking that means webform submissions are not correctly filtered, so if a user has other webform submissions (e.g. edit my details) it will incorrectly assume they have completed onboarding.

## Before

If a user has "edit my details" submissions, but no onboarding webform submissions they won't be redirected to the onboarding form on login.

## After

The user will be redirected to the onboarding form if they haven't completed it, regardless of other webform submissions.

---

- [x] Tests Pass

### CODE REVIEW

File: civihr_employee_portal/civihr_employee_portal.module [hunk 1]
```diff
@@ -5033,10 +5033,14 @@ function civihr_employee_portal_user_login(&$edit, $account) {
...
 function _civihr_employee_portal_should_do_onboarding($account) {
   $formName = OnboardingWebForm::NAME;
-  $nodes = node_load_multiple(NULL, ['title' => $formName]);
-  $node = current($nodes);
-  $canViewOnboardingForm = node_access('view', $node, $account);
-  $submissions = WebformHelper::getUserSubmissionsByName($account, $formName);
+  $onboardingWebform = WebformHelper::findOneByTitle($formName);
+
+  if (!$onboardingWebform) {
+    throw new \Exception("Could not find onboarding webform");
```
Comment (davialexandre) [inline civihr_employee_portal/civihr_employee_portal.module]: You can use single quotes here

File: civihr_employee_portal/src/Helpers/WebformHelper.php [hunk 2]
```diff
@@ -7,6 +7,21 @@
 class WebformHelper {
 
   /**
+   * @param $title
```
Comment (davialexandre) [inline civihr_employee_portal/src/Helpers/WebformHelper.php]: Missing type here

Comment (davialexandre) [inline civihr_employee_portal/src/Helpers/WebformHelper.php]: Please add a description to this method

File: civihr_employee_portal/src/Helpers/WebformHelper.php [hunk 3]
```diff
@@ -30,22 +45,25 @@ public static function getWebformComponentsByName($node, $name) {
...
+    $webformId = $webform->nid;
 
-    foreach ($submissions as $submission) {
-      $title = property_exists($submission, 'title') ? $submission->title : NULL;
-      if ($title === $name) {
-        $submissions[] = $submission;
+    foreach ($allUserSubmissions as $submission) {
+      $nodeID = property_exists($submission, 'nid') ? $submission->nid : NULL;
+      if ($webformId === $nodeID) {
+        $targetFormSubmissions[] = $submission;
```
Comment (davialexandre) [inline civihr_employee_portal/src/Helpers/WebformHelper.php]: Perhaps I'm slow today, but I can't see what is the difference here. Before we would only return the submissions with the same title as the given name, now we get the webform by title then return the submissions where the node ID is the same as the webform ID. Would you please give more details about this?

Comment (mickadoo) [reply civihr_employee_portal/src/Helpers/WebformHelper.php]: I'm confused myself, when I was debugging this I couldn't find any submission which had the property 'title', must have just been a mistake by me from the beginning.

**Pull request merged**